python/tests/base_testcase.py

```diff
@@ -0,0 +1,63 @@
+#  Copyright 2015-2017 TWO SIGMA OPEN SOURCE, LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#
+'''
+    The base class code for all Flint unit tests
+'''
+import unittest
+from abc import ABCMeta, abstractclassmethod
+import utils as test_utils
+from test_data import FORECAST_DATA, PRICE_DATA, VOL_DATA, VOL2_DATA, VOL3_DATA, INTERVALS_DATA
+
+
+class BaseTestcase(unittest.TestCase, metaclass=ABCMeta):
+    ''' Abstract base class for all Flint tests
+    '''
+    @abstractclassmethod
+    def setUpClass(cls):
+        ''' The automatic setup method for subclasses '''
+        return
+
+    @abstractclassmethod
+    def tearDownClass(cls):
+        ''' The automatic tear down method for subclasses '''
+        return
+
+    def forecast(self):
+        return self.flintContext.read.pandas(test_utils.make_pdf(FORECAST_DATA, ["time", "id", "forecast"]))
```

Comment:
The data is immutable so we don't need to create it every time it's needed. 

I think you can use the Memoize decorator to cache the result.